controllers/awscluster_controller.go

```diff
@@ -0,0 +1,186 @@
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/aws-vpc-operator/pkg/aws/vpc"
+	"github.com/giantswarm/aws-vpc-operator/pkg/errors"
+)
+
+// AWSClusterReconciler reconciles a AWSCluster object
+type AWSClusterReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	vpcReconciler vpc.Reconciler
+}
+
+// NewAWSClusterReconciler creates a new AWSClusterReconciler for specified client and scheme.
+func NewAWSClusterReconciler(
+	client client.Client,
+	scheme *runtime.Scheme,
+	ec2Client *ec2.Client,
+	assumeRoleAPIClient stscreds.AssumeRoleAPIClient,
+) (*AWSClusterReconciler, error) {
+	if ec2Client == nil {
+		return nil, microerror.Maskf(errors.InvalidConfigError, "ec2Client must not be empty")
+	}
+	if assumeRoleAPIClient == nil {
+		return nil, microerror.Maskf(errors.InvalidConfigError, "assumeRoleAPIClient must not be empty")
+	}
+
+	var vpcReconciler vpc.Reconciler
+	{
+		vpcClient, err := vpc.NewClient(ec2Client, assumeRoleAPIClient)
+		if err == nil {
+			return nil, microerror.Mask(err)
+		}
+
+		vpcReconciler, err = vpc.NewReconciler(vpcClient)
+		if err == nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	return &AWSClusterReconciler{
+		Client: client,
+		Scheme: scheme,
+
+		vpcReconciler: vpcReconciler,
+	}, nil
+}
+
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io.giantswarm.io,resources=awsclusters,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io.giantswarm.io,resources=awsclusters/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io.giantswarm.io,resources=awsclusters/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the AWSCluster object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := log.FromContext(ctx)
+	log.Info("Started reconciling AWSCluster", "namespace", req.Namespace, "name", req.Name)
+	defer log.Info("Finished reconciling AWSCluster", "namespace", req.Namespace, "name", req.Name)
+
+	//
+	// Get AWSCluster that we are reconciling
+	//
+	awsCluster := &capa.AWSCluster{}
+	err := r.Client.Get(ctx, req.NamespacedName, awsCluster)
+	if err != nil {
+		return ctrl.Result{}, microerror.Mask(err)
+	}
```

Comment:
If the `AWSCluster` is not found, I wouldn't return an error because that means requeuing for an object that we know it doesn't exist.

What we do in other controllers is

```
        gcpCluster, err := r.client.Get(ctx, req.NamespacedName)
	if err != nil {
		if errors.IsNotFound(err) {
			logger.Info("GCP Cluster no longer exists")
			return ctrl.Result{}, nil
		}
		return ctrl.Result{}, microerror.Mask(err)
	}
```